### PR TITLE
Merge mkinitcpio libalpm scripts into one

### LINKED ISSUE #10: Merge mkinitcpio libalpm scripts into one
Merge mkinitcpio-install and mkinitcpio-remove into one script that receives the either a install or remove option on the cmdline.

### PR DESCRIPTION
**Pull request by WhyNotHugo**
Mostly did this trying to familliarise myself better with how these scripts work.

Fixes #10

### CODE REVIEW

**Comment by WhyNotHugo** [comment]
Lemme know if this would need any refinement or if it's no longer desirable.

I do think it can be further refined/improved if the general idea is good.

File: libalpm/scripts/mkinitcpio [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+remove_kernel() {
+    # remove the actual kernel and images for the package being removed
+    kernel="/boot/vmlinuz-${1}"
+    preset="/etc/mkinitcpio.d/${1}.preset"
+    initramfs="/boot/initramfs-${1}.img"
+    fallback_initramfs="/boot/initramfs-${1}-fallback.img"
+    if [[ -e $kernel ]]; then
+        # remove the installed kernel
+        rm $kernel
+    fi
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:49]:
```suggestion
    # remove the installed kernel
    rm -f -- $kernel
```

File: libalpm/scripts/mkinitcpio [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    process_preset "${1}" $preset
+
+    if [[ -e $initramfs ]]; then
+        # remove the main image
+        rm $initramfs
+    fi
+    if [[ -e $fallback_initramfs ]]; then
+        # remove the fallback image
+        rm $fallback_initramfs
+    fi
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:60]:
```suggestion
    # remove images
    rm -f -- $initramfs $fallback_initramfs
```

File: libalpm/scripts/mkinitcpio [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    if ! read -r pkgbase > /dev/null 2>&1 < "${line%/vmlinuz}/pkgbase"; then
+        # if the kernel has no pkgbase, we skip it
+        continue
+    fi
+
+    if [ "$1" = "install" ]; then
+      install_kernel "$pkgbase"
+    elif [ "$1" = "remove" ]; then
+      remove_kernel "$pkgbase"
+    fi
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:79]:
```suggestion
    case "$1" in
        install) install_kernel "$pkgbase";;
        remove) remove_kernel "$pkgbase";;
    esac
```

File: libalpm/scripts/mkinitcpio [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      args=(-P)
+    elif [ "$1" = "remove" ]; then
+      shopt -s nullglob
+      for preset in /etc/mkinitcpio.d/*.preset; do
+          pkgbase=${preset##*/}
+          pkgbase=${pkgbase%.preset}
+          process_preset "${pkgbase}" "$preset"
+      done
+      shopt -u nullglob
+    fi
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:94]:
```suggestion
    case "$1" in
        install)
            # change to use all presets
            args=(-P)
            ;;
        remove)
          shopt -s nullglob
          for preset in /etc/mkinitcpio.d/*.preset; do
              pkgbase=${preset##*/}
              pkgbase=${pkgbase%.preset}
              process_preset "${pkgbase}" "$preset"
          done
          shopt -u nullglob
          ;;
    esac
```

File: libalpm/scripts/mkinitcpio [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+fi
+
+if [ "$1" = "install" ] && (( ${#args[@]} )); then
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:97]:
```suggestion
if [[ "$1" == "install" ]] && (( ${#args[@]} )); then
```

**Review by nl6720 — CHANGES_REQUESTED** [review]

File: libalpm/scripts/mkinitcpio [hunk 6]
```diff
@@ -0,0 +1,99 @@
+#!/bin/bash -e
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:1]:
```suggestion
#!/usr/bin/env bash

 set -e
```

**Review by nl6720 — CHANGES_REQUESTED** [review]

**Comment by WhyNotHugo** [comment]
All these make perfect sense, applied.

**Review by nl6720 — CHANGES_REQUESTED** [review]
Please quote all variables.

File: libalpm/scripts/mkinitcpio [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+args=()
+package=0
+
+process_preset() {
+    if [[ -n "${pkgbase}" && -e $preset ]]; then
+        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
+            if [[ ! -e $preset.pacsave ]]; then
+                # save the preset as pacsave
+                mv "$preset" "$preset.pacsave" && return 0
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:13]: Terminate option parsing when variables are involved.

```suggestion
                mv -- "$preset" "$preset.pacsave" && return 0
```

File: libalpm/scripts/mkinitcpio [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+            fi
+        else
+            # remove the preset
+            rm "$preset" && return 0
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:17]:
```suggestion
            rm -- "$preset" && return 0
```

File: libalpm/scripts/mkinitcpio [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        fi
+    fi
+}
+
+install_kernel() {
+    preset="/etc/mkinitcpio.d/${1}.preset"
+    if [[ ! -e $preset ]]; then
+        if [[ -e $preset.pacsave ]]; then
+            # move the pacsave to the template
+            mv "${preset}.pacsave" "$preset"
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:27]:
```suggestion
            mv -- "${preset}.pacsave" "$preset"
```

File: libalpm/scripts/mkinitcpio [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    rm -f -- "$kernel"
+
+    process_preset "${1}" "$preset"
+
+    # remove images
+    rm -f -- "$initramfs" "$fallback_initramfs"
+}
+
+while read -r line; do
+    if [[ $line != */vmlinuz ]]; then
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:58]: Quote:

```suggestion
    if [[ "$line" != */vmlinuz ]]; then
```

**Review by nl6720 — CHANGES_REQUESTED** [review]

File: libalpm/scripts/mkinitcpio (hunk 9 above)
Comment (WhyNotHugo) [reply libalpm/scripts/mkinitcpio]: This never occurred to me. Is should never be an issue in this context, but still a good idea.

File: libalpm/scripts/mkinitcpio [hunk 11]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+ set -e
+
+args=()
+package=0
+
+process_preset() {
+    if [[ -n "${pkgbase}" && -e $preset ]]; then
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:9]:
```suggestion
    if [[ -n "$pkgbase" && -e "$preset" ]]; then
```

File: libalpm/scripts/mkinitcpio [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+            # remove the preset
+            rm -- "$preset" && return 0
+        fi
+    fi
+}
+
+install_kernel() {
+    preset="/etc/mkinitcpio.d/${1}.preset"
+    if [[ ! -e $preset ]]; then
+        if [[ -e $preset.pacsave ]]; then
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:25]:
```suggestion
    if [[ ! -e "$preset" ]]; then
        if [[ -e "$preset.pacsave" ]]; then
```

File: libalpm/scripts/mkinitcpio [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        install)
+            # change to use all presets
+            args=(-P)
+            ;;
+        remove)
+          shopt -s nullglob
+          for preset in /etc/mkinitcpio.d/*.preset; do
+              pkgbase=${preset##*/}
+              pkgbase=${pkgbase%.preset}
+              process_preset "${pkgbase}" "$preset"
```
Comment (nl6720) [inline libalpm/scripts/mkinitcpio:86]:
```suggestion
              process_preset "$pkgbase" "$preset"
```

**Comment by nl6720** [comment]
Once the few minor style issues are addressed, it looks good to merge.

**Review by nl6720 — APPROVED** [review]
Please squash the commits.

**Review by Foxboron — APPROVED** [review]

**Pull request merged**